Plugins/MultiplayerSessionPlugin/Source/MultiplayerSessionPlugin/Private/MultiplayerSessionSubsystem.cpp

```diff
@@ -20,36 +21,139 @@ UMultiplayerSessionSubsystem::UMultiplayerSessionSubsystem():
 
 void UMultiplayerSessionSubsystem::CreateSession(int32 NumPublicConnections, FString MatchType)
 {
+	// kill existing session if present, then make a new one
+	// get player or world
+	// go to lobby
+	// add create session complete to delegate list
+
+	// kill existing session if present, then make a new one
+	if (!SessionInterface.IsValid())
+	{
+		return;
+	}
+
+	auto ExistingSession = SessionInterface->GetNamedSession(NAME_GameSession);
+	if (ExistingSession != nullptr)
+	{
+		SessionInterface->DestroySession(NAME_GameSession);
+	}
+
+	// storing delegate in handler for removal during teardown
+	CreateSessionCompleteDelegateHandle = SessionInterface->AddOnCreateSessionCompleteDelegate_Handle(CreateSessionCompleteDelegate);
+
+	LastSessionSettings = MakeShareable(new FOnlineSessionSettings());
+	LastSessionSettings->bIsLANMatch = IOnlineSubsystem::Get()->GetSubsystemName() == "NULL" ? true : false;
```

Comment:
TODO: Need to figure out what I want to do with dev builds for this. Not important now, but this isn't doin it for me.